src/postgres-util/src/desc.proto

```diff
@@ -11,19 +11,29 @@ syntax = "proto3";
 
 package mz_postgres_util.desc;
 
+message ProtoPostgresKeyDesc {
+    uint32 oid = 1;
+    string name = 2;
+    repeated uint32 cols = 3;
+    bool is_primary = 4;
+}
+
 message ProtoPostgresTableDesc {
     string name = 1;
     string namespace = 2;
     uint32 oid = 3;
     repeated ProtoPostgresColumnDesc columns = 4;
+    repeated ProtoPostgresKeyDesc keys = 5;
 }
 
 message ProtoPostgresColumnDesc {
+    reserved 5;
+    reserved "primary_key";
     string name = 1;
     uint32 type_oid = 2;
     int32 type_mod = 3;
     bool nullable = 4;
-    bool primary_key = 5;
```

Comment:
nit: I think the recommended Protobuf practice is to [mark this field as `reserved`](https://protobuf.dev/programming-guides/proto3/#reserved).

See your changes to `ProtoRecord` and `ProtoMap`. Is there a reason to diverge from this scheme here?

Reply:
Yes, I already did that reservation here: https://github.com/MaterializeInc/materialize/pull/18403/files/117fabb46e9bd932b900f751e5f287b421fe50b7#diff-293f6b79477ec6898d37dd546f51070b82779adcf2ffa13695a460ea9c2223a3R31

> See your changes to ProtoRecord and ProtoMap. Is there a reason to diverge from this scheme here?

Are you referring to this diverging from `ProtoColumnType`? If so, the issue is that we have no way to convert from  OID to `ScalarType` in a dataflow. If not, can I get some clarity about what this refers to?